### PR TITLE
Added ProjectileCollideEvent and entity and projectile event values

### PR DESCRIPTION
**Pull request by DeltaRays**
### Description

The title says everything
---
**Target Minecraft Versions:**   1.12+  
**Requirements:**   Paper  
**Related Issues:**

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -464,6 +466,25 @@ public Entity get(final ProjectileLaunchEvent e) {
 				return e.getEntity();
 			}
 		}, 0, "Use 'projectile' and/or 'shooter' in shoot events", ProjectileLaunchEvent.class);
+		//ProjectileCollideEvent
+		if (Skript.classExists("com.destroystokyo.paper.event.entity.ProjectileCollideEvent")) {
+			if (Skript.methodExists(ProjectileCollideEvent.class, "getEntity"))
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:471]: Are these method checks necessary? If they are, you can just ignore this.

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 2]
```diff
@@ -338,6 +339,12 @@
 						"\tevent-projectile is arrow",
 						"\tdelete event-projectile")
 				.since("1.0");
+		
+		if(Skript.classExists("com.destroystokyo.paper.event.entity.ProjectileCollideEvent")) Skript.registerEvent("Projectile Collide", SimpleEvent.class, ProjectileCollideEvent.class, "projectile collide")
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:343]: More of a stylistic change, but you can put the check and registration on separate lines? Helps with readability and avoiding long lines 😃

Comment (DeltaRays) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:343]: Done 👍

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 3]
```diff
@@ -1,18 +1,18 @@
 /**
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:1]: Looks like your IDE reformatted the file. Are you able to revert this change?

Comment (DeltaRays) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:1]: I actually formatted it on purpose, is there an issue with that?

Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:1]: Overall, no, but it's a bit out of the scope of this specific PR, so it might be better to do in a separate one.

Comment (DeltaRays) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:1]: Okay

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 4]
```diff
@@ -118,13 +118,15 @@
...
 
+import com.destroystokyo.paper.event.entity.ProjectileCollideEvent;
 import com.destroystokyo.paper.event.player.PlayerArmorChangeEvent;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.aliases.Aliases;
 import ch.njol.skript.aliases.ItemType;
 import ch.njol.skript.command.CommandEvent;
 import ch.njol.skript.events.EvtMoveOn;
+import ch.njol.skript.lang.util.SimpleEvent;
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:129]: You don't need this import

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java (hunk 1 above)
Comment (Wealthyturtle) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:471]: These method checks are not necessary, they've been there since paper implemented this event. https://github.com/PaperMC/Paper/blob/ver/1.11.2/Spigot-API-Patches/0042-Add-ProjectileCollideEvent.patch

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 2 above)
Comment (Wealthyturtle) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:342]: Not done?

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 5]
```diff
@@ -337,6 +338,12 @@
 						"\tevent-projectile is arrow",
 						"\tdelete event-projectile")
 				.since("1.0");
+		
+		if(Skript.classExists("com.destroystokyo.paper.event.entity.ProjectileCollideEvent")) Skript.registerEvent("Projectile Collide", SimpleEvent.class, ProjectileCollideEvent.class, "projectile collide")
+			.description("Called when a projectile collides with an entity")
+			.examples("on projectile collide:",
+				"\tteleport shooter of event-projectile to event-entity")
+			.since("INSERT VERSION");
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:346]: Also missing a ``.requiredPlugins("Paper")``

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 2 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:342]: The more I use git the less I understand it

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java (hunk 1 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java]: Removed

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java (hunk 4 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java]: Removed

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 5 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:348]: Added it

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 6]
```diff
@@ -337,6 +338,14 @@
 						"\tevent-projectile is arrow",
 						"\tdelete event-projectile")
 				.since("1.0");
+		
+		if(Skript.classExists("com.destroystokyo.paper.event.entity.ProjectileCollideEvent"))
+			Skript.registerEvent("Projectile Collide", SimpleEvent.class, ProjectileCollideEvent.class, "projectile collide")
+			.description("Called when a projectile collides with an entity")
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:344]: Just another full stop at this description and I think that's about it.

**Comment by Wealthyturtle** [comment]
@DeltaRays please merge master into your branch for this PR to be merged
